api/v1alpha1/deviceconfig_types.go

```diff
@@ -81,3 +84,15 @@ func (dc *DeviceConfig) GetNodeSelector() map[string]string {
 	}
 	return ns
 }
+
+func (dc *DeviceConfig) GetGaudiNodeSelector() map[string]string {
```

Comment:
What do you think about replacing the node label [here](https://github.com/fabiendupont/habana-ai-operator/blob/main/api/v1alpha1/deviceconfig_types.go#L80) with the `fmt.Sprintf("habana.ai/hpu.%s.present", deviceType)` you provided in the method, instead of creating a new method and an new API field (i.e. the `GaudiNodeSelector` field)?

Reply:
can the GetNodeSelector functions get parameters? in that case I will replace my new function with a switch inside the function

Reply:
Absolutely, you can define any parameters you need for the `GetNodeSelector` method. 

Just to give you more context about this method, the main reason for having it in the `DeviceConfig` CRD, instead as a helper method/function in the controller or elsewhere in the codebase, is that it explicitly refers to a field (i.e. the `NodeSelector`) of that CRD and its default values. I would say we can do whatever we think it's best regarding its parameters, as long as we don't code too much controller business logic in there and we keep it related to the fields of the CRD.